src/core/typing.ts

```diff
@@ -46,21 +43,24 @@ export interface Typing extends EmitsDiscontinuities {
    * Get the current typers, a set of clientIds.
    * @returns A Promise of a set of clientIds that are currently typing.
    */
-  get(): Promise<Set<string>>;
+  get(): Set<string>;
 
   /**
-   * Start indicates that the current user is typing. This will emit a typingStarted event to inform listening clients and begin a timer,
-   * once the timer expires, a typingStopped event will be emitted. The timeout is configurable through the typingTimeoutMs parameter.
-   * If the current user is already typing, it will reset the timer and being counting down again without emitting a new event.
+   * Start indicates that the current user is typing.
+   * This will emit a `typing.start` event to inform listening clients and begin a heartbeat timer,
+   * which can be configured through the `heartbeatIntervalMs` parameter.
```

Comment:
We probably don't need to describe the timer/internal workings in the public API, perhaps something as simple as:

"This will send a `typing.started` event to the server. Events are throttled according to the `heartbeatIntervalMs` room option. If an event has been sent within the interval, this operation is no-op."